app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginFragment.java

```diff
@@ -159,12 +158,17 @@ public void OnForgotPassword() {
         Toast.makeText(getContext(), "Mail is Sent", Toast.LENGTH_SHORT).show();
     }
 
+    @Override
+    public void ShowLoading() {
```

Comment:
Now it's better 👍 
For further improvements, also, you can create a new callback to `hideLoading` and keep dismiss the dialog there instead of dismissing the loading inside `OnPasswordIsError` and `OnMailIsError`... keep it more single responsible